app/src/main/java/ch/epfl/sweng/swenggolf/offer/ListAnswerAdapter.java

```diff
@@ -0,0 +1,143 @@
+import ch.epfl.sweng.swenggolf.database.ValueListener;
+import ch.epfl.sweng.swenggolf.tools.ThreeFieldsViewHolder;
+
+public class ListAnswerAdapter extends RecyclerView.Adapter<ListAnswerAdapter.AnswerViewHolder> {
+    private Answers answers;
+    private Offer offer;
+    private static final int HEART_FULL = R.drawable.ic_favorite;
+    private static final int HEART_EMPTY = R.drawable.ic_favorite_border;
+
+    public static class AnswerViewHolder extends ThreeFieldsViewHolder {
+
+        public AnswerViewHolder(View view) {
+            super(view, R.id.user_name, R.id.answer_description, R.id.user_pic);
+        }
+    }
+
+    /**
+     * Constructs a ListAnswerAdapter for the RecyclerView.
+     *
+     * @param answers the objet containing the list of answers to be displayed
+     */
+    public ListAnswerAdapter(Answers answers, Offer offer) {
+        if (answers == null || answers.getAnswerList() == null || offer == null) {
+            throw new IllegalArgumentException();
+        }
+        this.answers = answers;
+        this.offer = offer;
+    }
+
+    /**
+     * Sets the answers field.
+     * @param answers the new answers
+     */
+    public void setAnswers(Answers answers) {
+        this.answers = answers;
+        notifyDataSetChanged();
+    }
+
+    /**
+     * Gets the answers.
+     * @return the answers
+     */
+    public Answers getAnswers() {
+        return answers;
+    }
+
+
+    @Override
+    public AnswerViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.reactions_others, parent, false);
+        return new AnswerViewHolder(view);
+    }
+
+    // Replace the contents of a view (invoked by the layout manager)
+    @Override
+    public void onBindViewHolder(final AnswerViewHolder holder, int position) {
+        final Answer answer = answers.getAnswerList().get(position);
+
+        // get the user data from database
+        ValueListener<User> vlUser = new ValueListener<User>() {
+            @Override
+            public void onDataChange(User value) {
+                TextView userName = (TextView) holder.getFieldOne();
+                userName.setText(value.getUserName());
+                userName.setContentDescription(
+                        "username"+Integer.toString(holder.getAdapterPosition()));
+                ImageView userPic = (ImageView) holder.getFieldThree();
+                Picasso.with(userPic.getContext())
+                        .load(Uri.parse(value.getPhoto()))
+                        .placeholder(R.drawable.gender_neutral_user1)
+                        .fit().into(userPic);
+                userPic.setContentDescription("pic"+Integer.toString(holder.getAdapterPosition()));
+            }
+
+            @Override
+            public void onCancelled(DbError error) {
+                Log.d(error.toString(), "Unable to load user from database");
```

Comment:
Do we want to tell the user that something went wrong?
In that case, a `Toast` might be an option, otherwise, this is fine :)

Reply:
Yes, you're absolutely right, I thought about it recently. If you agree, we could push this for the moment and I will add a message that the load couldn't be done and test it, and that will be done in a few days, because I also want to add another thing.

Reply:
Sure, sounds good!